### PR TITLE
Qbft validator selection was incorrectly being applied to non validator selection mode transitions

### LINKED ISSUE #2894: Fix QBFT header validation rules are being applied incorrectly for non validator selection mode transitions
The new QBFT transition for changing the validator selection mode does use the correct header validation rules if there is a non validator selection transition.

E.g. Starting in contract validator mode and changing the block reward at block 5. From block onwards the incorrect block header validation rules are being used. This results in the chain stalling.

```
"config": {
"chainid": 2017,
"homesteadBlock": 0,
"eip150Block": 0,
"eip155Block": 0,
"eip158Block": 0,
"byzantiumBlock": 0,
"qbft": {
	"epochlength": 30000,
	"blockperiodseconds": 5,
	"requesttimeoutseconds": 10,
	"validatorcontractaddress": "0x0000000000000000000000000000000000008888"
},
"transitions": {
	"qbft": [
		{
			"block": 5,
			"blockrewardwei": 20
		}
	]
}
```

### PR DESCRIPTION
**Pull request by jframe**
## PR description
Qbft validation selection was incorrectly being applied for non validator selection mode transitions. This was caused by the QbftProtocolSchedule not considering the current validator selection mode. The BftForksSchedule has been added to the QbftProtocolSchedule so that current mode is used even when the transition is not for changing the validation selection mode.

fixes #2894

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/QbftBlockHeaderUtils.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package org.hyperledger.besu.consensus.qbft;
```
Comment (usmansaleem) [inline consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/QbftBlockHeaderUtils.java:1]: license header

**Review by usmansaleem — APPROVED** [review]
LGTM

File: consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/QbftBlockHeaderUtils.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        number, proposerNodeKey, parent, modifier, builder, qbftExtraDataEncoder);
+
+    final BftExtraData bftExtraData =
+        BftExtraDataFixture.createExtraData(
+            builder.buildHeader(),
+            Bytes.wrap(new byte[BftExtraDataCodec.EXTRA_VANITY_LENGTH]),
+            Optional.empty(),
+            Collections.emptyList(),
+            singletonList(proposerNodeKey),
+            0x2A,
```
[the remark this answers is not in the document]
Comment (jframe) [reply consensus/qbft/src/test/java/org/hyperledger/besu/consensus/qbft/QbftBlockHeaderUtils.java]: It's an arbitrary value for the round number, i've added a constant to make this clearer.

**Review by usmansaleem — APPROVED** [review]
LGTM

**Pull request merged**